plaid/errors.py

```diff
@@ -11,14 +11,35 @@ class PlaidError(Exception):
                                         for programmatic use. May be None.
     :ivar   str     request_id:         A unique id returned for all server
                                         responses.
+    :ivar   list    causes:             A list of reasons explaining why the
+                                        error happened.
     '''
 
-    def __init__(self, message, type, code, display_message, request_id=""):
+    def __init__(
+        self,
+        message,
+        type,
+        code,
+        display_message,
+        request_id="",
+        causes=[],
+    ):
         super(PlaidError, self).__init__(message)
         self.type = type
         self.code = code
         self.display_message = display_message
         self.request_id = request_id
+        self.causes = []
+
+        for cause in causes:
+            c = PlaidCause(
+                cause['item_id'],
+                cause['error_message'],
+                cause['error_type'],
+                cause['error_code'],
+                cause['display_message'] if 'display_message' in cause else '',
+            )
+            self.causes.append(c)
```

Comment:
nit: can we make this a list-comprehension?

Reply:
⚡️